page_objects/sidebar.py

```diff
@@ -0,0 +1,64 @@
+from page_objects.license_details_page import LicenseDetailsPage
+from page_objects.notifications_page import NotificationsPage
+from utilities.ui_utilities.base_page import BasePage
+from selenium.webdriver.common.by import By
+
+
+class SideBar(BasePage):
+    def __init__(self, driver):
+        super().__init__(driver)
+
+    __login_button = (By.XPATH, "//*[@class='sc-ifAKCX sc-sdtwF lGUCz']")
+    __logout_button = (By.XPATH, "//span[@aria-label='Logout']")
+    __notifications_sidebar_item = (By.XPATH, "//div/b[contains(text(), 'Notifications')]")
+    __license_details_sidebar_item = (By.XPATH, "//div/b[contains(text(), 'License Details')]")
+    __github_link = (By.XPATH, "//*[@class='sc-dyGzUR cbOCWZ']")
+    __sidebar_area = (By.XPATH, "//nav")
+    __collapse_button = (By.XPATH, "//*[@data-icon='arrow-circle-left']")
+    __expand_button = (By.XPATH, "//*[@data-icon='arrow-circle-right']")
+
+    def is_logout_present(self):
+        return self.is_displayed(self.__logout_button)
+
+    def click_logout(self):
+        self.click(self.__logout_button)
+        return self
+
+    def is_login_present(self):
+        return self.is_displayed(self.__login_button)
+
+    def go_to_notifications_page(self):
+        self.click(self.__notifications_sidebar_item)
+        return NotificationsPage(self._driver)
+
+    def get_github_link(self):
+        self.is_displayed(self.__github_link)
+        return self.get_link_value(self.__github_link)
+
+    def hover_sidebar(self):
+        self.hover_the_element(self.__sidebar_area)
+        return self
```

Comment:
what is the purpose of this function? you can use just function from BasePage where needed

Reply:
1. without separate hover, collapse/extend button does not work
2. in case of triggering hover from BasePage, locator should be passed, but it's private inside respective PageObject.

Reply:
So add this hover to the collapse/expand button. No need for overriding function here

Reply:
fixed.